samples/node/pub_sub/index.ts

```diff
@@ -0,0 +1,194 @@
+            type: 'string',
+            required: false
+        })
+        .option('cert', {
+            alias: 'c',
+            description: 'FILE: path to a PEM encoded certificate to use with mTLS',
+            type: 'string',
+            required: false
+        })
+        .option('key', {
+            alias: 'k',
+            description: 'FILE: Path to a PEM encoded private key that matches cert.',
+            type: 'string',
+            required: false
+        })
+        .option('client_id', {
+            alias: 'C',
+            description: 'Client ID for MQTT connection.',
+            type: 'string',
+            required: true
+        })
+        .option('topic', {
+            alias: 't',
+            description: 'STRING: Targeted topic',
+            type: 'string',
+            default: 'sdk/test/NodeJSv2'
+        })
+        .option('count', {
+            alias: 'n',
+            default: 10,
+            description: 'Number of messages to publish/receive before exiting. ' +
+            'Specify 0 to run forever.',
+            type: 'number',
+            required: false
+        })
+        .option('use_websocket', {
+            alias: 'W',
+            default: false,
+            description: 'To use a websocket instead of raw mqtt. If you ' +
+            'specify this option you must specify a region for signing, you can also enable proxy mode.',
+            type: 'boolean',
+            required: false
+        })
+        .option('signing_region', {
+            alias: 's',
+            default: 'us-east-1',
+            description: 'If you specify --use_websocket, this ' +
+            'is the region that will be used for computing the Sigv4 signature',
+            type: 'string',
+            required: false
+        })
+        .option('proxy_host', {
+            alias: 'H',
+            description: 'Hostname for proxy to connect to. Note: if you use this feature, ' +
+            'you will likely need to set --ca_file to the ca for your proxy.',
+            type: 'string',
+            required: false
+        })
+        .option('proxy_port', {
+            alias: 'P',
+            default: 8080,
+            description: 'Port for proxy to connect to.',
+            type: 'number',
+            required: false
+        })
+        .option('message', {
+            alias: 'M',
+            description: 'Message to publish.',
+            type: 'string',
+            default: 'Hello world!'
+        })
+        .option('verbosity', {
+            alias: 'v',
+            description: 'BOOLEAN: Verbose output',
+            type: 'string',
+            default: 'none',
+            choices: ['fatal', 'error', 'warn', 'info', 'debug', 'trace', 'none']
+        })
+        .help()
+        .alias('help', 'h')
+        .showHelpOnFail(false)
+}, main).parse();
+
+async function execute_session(connection: mqtt.MqttClientConnection, argv: Args) {
```

Comment:
This doesn't use anything from the IoT service client. Should it?

Reply:
nope, greengrass does, and we can/should add more samples soon.